.github/workflows/retry.yml

```diff
@@ -0,0 +1,184 @@
+# Please ping @DilumAluthge when making any changes to this file.
+
+# Here are some steps that we take in this workflow file for security reasons:
+# 1. We do not checkout any code.
+# 2. We do not run any external actions.
+# 3. We do not give the `GITHUB_TOKEN` any permissions.
+# 4. We only give the Buildkite API token (`BUILDKITE_API_TOKEN_RETRY`) the minimum necessary
+#    set of permissions.
+
+# Important note to Buildkite maintainers:
+# In order to make this work, you need to tell Buildkite that it should NOT create a brand-new
+# build when someone closes and reopens a pull request. To do so:
+# 1. Go to the relevant pipeline (e.g. https://buildkite.com/julialang/julia-master).
+# 2. Click on the "Pipeline Settings" button.
+# 3. In the left sidebar, under "Pipeline Settings", click on "GitHub".
+# 4. In the "GitHub Settings", under "Build Pull Requests", make sure that the "Skip pull
+#    request builds for existing commits" checkbox is checked. This is the setting that tells
+#    Buildkite that it should NOT create a brand-new build when someone closes and reopens a
+#    pull request.
+# 5. At the bottom of the page, click the "Save GitHub Settings" button.
+
+name: Retry Failed Buildkite Jobs
+
+on:
+  # When using the `pull_request_target` event, all PRs will get access to secret environment
+  # variables (such as the `BUILDKITE_API_TOKEN_RETRY` secret environment variable), even if
+  # the PR is from a fork. Therefore, for security reasons, we do not checkout any code in
+  # this workflow.
+  # TODO: change `pull_request` to `pull_request_target`.
+  pull_request:
+
+    # TODO: delete the following line (once we have completely transitioned from Buildbot to Buildkite)
+    types: [ reopened, labeled ]
+
+    # TODO: uncomment the following line (once we have completely transitioned from Buildbot to Buildkite)
+    # types: [ reopened ]
+
+# We do not give the `GITHUB_TOKEN` any permissions.
+permissions:
+  statuses: none
+
+jobs:
+  retry:
+    name: retry
+    runs-on: ubuntu-latest
+
+    # TODO: delete the following line (once we have completely transitioned from Buildbot to Buildkite)
```

Comment:
Is this still relevant?  I thought this PR was buildkite-specific and has nothing to do with buildbot.